zendesk/user.go

```diff
@@ -75,6 +75,16 @@ func (c *client) CreateUser(user *User) (*User, error) {
 	return out.User, err
 }
 
+// UpsertUser creates or updates a user.
+//
+// Zendesk Core API docs: https://developer.zendesk.com/rest_api/docs/core/users#create-or-update-user
+func (c *client) UpsertUser(user *User) (*User, error) {
```

Comment:
I would name it `CreateOrUpdateUser(user *User)` just to be consistent with the naming in the docs.